libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -694,10 +694,10 @@ namespace webassembly {
           *                the sync call. LSB bit indicates read-only. All other bits
           *                are reserved to be 0.
           * @param data - the data of the sync call, which may include function name, arguments and other information.
-          * @return the number of bytes of the return value of the call (to be used by next get_call_return_value) .
+          * @return the number of bytes of the return value of the call (to be used by next get_call_return_value) if succeeded, return -1 if sync_call entry point does not exist or has invalid signature .
           *
          */
-         uint32_t call(name receiver, uint64_t flags, span<const char> data);
+         int32_t call(name receiver, uint64_t flags, span<const char> data);
```

Comment:
I know this is academic but it's the cost of `CONFIGURABLE_WASM_LIMITS2`'s infinite knobs: `int32_t` isn't enough for greater than 2GB unless the contract treats this as unsigned. Unless there is a compelling reason to keep it 32-bit I'd suggest moving to `int64_t` so that the semantics are clearer for these academic edge cases.

Reply:
The value returned by `call()` (the number of sync call return value) is used as the size component for constructing `memory` argument in `get_call_return_value(span<char> memory)`. In WASM, the memory span is represented as  (param i32 i32). Is it safe to cast down from i64 to i32 when the user uses it?

Reply:
Another point: the return value size is limited by `max_sync_call_data_size`, which is likely less than `int32_t` size.

Reply:
@arhag Any comment about this?

Reply:
I don't really have a position on this one.

Reply:
To prevent any potential edge cases, I changed to return type to `int64_t`.

Thanks.

https://github.com/AntelopeIO/spring/pull/1266/commits/f6605d3abd44969205395e1d175f1ccbe552f945